internal/mode/static/handler.go

```
@@ -440,34 +499,36 @@ func getGatewayAddresses(
 	ctx context.Context,
 	k8sClient client.Client,
 	svc *v1.Service,
-	podConfig ngfConfig.GatewayPodConfig,
+	gateway *graph.Gateway,
+	gatewayClassName string,
 ) ([]gatewayv1.GatewayStatusAddress, error) {
-	podAddress := []gatewayv1.GatewayStatusAddress{
-		{
-			Type:  helpers.GetPointer(gatewayv1.IPAddressType),
-			Value: podConfig.PodIP,
-		},
+	if gateway == nil {
+		return nil, nil
```

If the gateway == nil, don't we want to return an error? Seems like the places that use the gatewayAddr only check if there is an error returned and assume the addr is not nil

---

oh i see, PrepareGatewayRequests filters our nil gateways so this isn't needed